### PR TITLE
Add Code Review checklist to PR Template

### PR DESCRIPTION
**Pull request by bgfoshay**
## What’s changing?

Adding a code review checklist to the PR template

## Why?

Taking steps to improve our development workflow

## This PR has:

- [ ] **Commented source code** (_required on public classes and methods, and elsewhere as appropriate_)

- [x] **Developer documentation** (_for new build/test/API changes or complex portions of the system_)

- [ ] **Automated unit tests** (_to maintain or increase level of code coverage_)

- [ ] **Changes to IAC** (_add any deployment steps that require manual intervention to the draft release notes_)

### CODE REVIEW

**Comment by dborgen2** [comment]
I think this is a great idea. Just to confirm, these checkboxes are for the reviewer to check, not the person who create the PR?

**Review by ryangeorgeharvey — COMMENTED** [review]
More than just listing the JIRA ticket, all branches should be named as the JIRA ticket. For example, NAC-1234 or NAC-1234-Email. This makes it easy to find branches/commits based on the JIRA ticket, and will allow for JIRA and GitHub Integration.

File: docs/pull_request_template.md [hunk 1]
```diff
@@ -16,4 +16,46 @@ _Link to specific issue (ex: Closes #123)_
...
+- [ ] Are there API changes?
+    - [ ] Does API make sense? Useful and concise?
+    - [ ] Are existing API interfaces/contracts intact? i.e. no breaking changes
+    - [ ] Are API tests included?
+    - [ ] Are internals kept private and not leaked into the API?
+- [ ] Are there new Azure/Infrastructure Resources or changes to existing resource configurations?
+[ ] IAC Scripts are readable & maintainable
+    - [ ] IAC Scripts are documented
+    - [ ] IAC scripts pass required checks
+    - [ ] IAC scripts run successfully
```
Comment (ryangeorgeharvey) [inline docs/pull_request_template.md:40]: Have IaC scripts run successfully against an existing deployed environment?
Have IaC scripts run successfully against a newly deployed environment? 
Do the IaC changes require deleting existing resources?

**Comment by ryangeorgeharvey** [comment]
> I think this is a great idea. Just to confirm, these checkboxes are for the reviewer to check, not the person who create the PR?

Hmm, this is an interesting question. I think most of these should be answered by the person requesting code review. Then the code reviewer should verify them.

**Comment by dborgen2** [comment]
> > I think this is a great idea. Just to confirm, these checkboxes are for the reviewer to check, not the person who create the PR?
> 
> Hmm, this is an interesting question. I think most of these should be answered by the person requesting code review. Then the code reviewer should verify them.

Makes sense. You could even have two checkboxes next to each item, one for the person requesting the review and one for the reviewer.

**Comment by bgfoshay** [comment]
> > > I think this is a great idea. Just to confirm, these checkboxes are for the reviewer to check, not the person who create the PR?
> > 
> > 
> > Hmm, this is an interesting question. I think most of these should be answered by the person requesting code review. Then the code reviewer should verify them.
> 
> Makes sense. You could even have two checkboxes next to each item, one for the person requesting the review and one for the reviewer.

That's an interesting idea. I'm open to that if others think it'd be useful to have a checkbox for both.

**Review by dborgen2 — APPROVED** [review]
It looks good to me. Thanks for putting this together, it should help a lot.

**Pull request merged**